openedx/core/djangoapps/user_api/views.py

```diff
@@ -345,10 +350,63 @@ def post(self, request):
             }
             return JsonResponse(errors, status=400)
 
+        password = data.get("password")
+        password_copy = data.get("password_copy")
+        if password and password_copy and password != password_copy:
+            errors = {
+                'password_copy': [{"user_message": "Passwords don't match."}]
+            }
+            return JsonResponse(errors, status=400)
+
         response = JsonResponse({"success": True})
         set_logged_in_cookies(request, response, user)
+
+        if getattr(settings, 'REGISTRATION_EMAIL_FULL_VERIFICATION', None) is not None:
```

Comment:
In our edx system we had many complaints from users: they mistyped their emails or passwords and successfully logged in LMS but couldn't log in a second time (and their work in 1st session was lost).